Automattic/wp-calypso
# Congrats: remove unused failed purchase components

Related to https://github.com/Automattic/dotcom-forge/issues/6757

## Proposed Changes
After discussion here p9Jlb4-7Hx-p2#comment-11302 we decided to remove all "failed purchase" related code since we don't support this flow yet, so it's totally redundant to keep it.

## Testing Instructions
1) Visual review and tests should suffice
2) Additionally you can try to make a few different purchases and assert that everything still works well
